x/stake/keeper/distribute.go

```diff
@@ -0,0 +1,127 @@
+package keeper
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+func (k Keeper) DistributeBeaconChainInBreathBlock(ctx sdk.Context, bscFee int64) {
+	for k.hasNextBatchRewards(ctx) {
+		k.distributeSingleBatch(ctx, MockSideChainIDForBeaconChain)
+	}
+
+	validators, height, found := k.GetHeightValidatorsByIndex(ctx, daysBackwardForValidatorSnapshot)
+	avgFee := bscFee / int64(len(validators))
+	changeFee := bscFee - avgFee*int64(len(validators))
+	if !found {
```

Comment:
Let us move the `if  !found` upwards to fail fast

Reply:
done